spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -216,6 +226,33 @@ protected void afterUnbindProducer(ProducerDestination destination, P producerPr
 				@Override
 				protected void afterUnbind() {
 					AbstractMessageChannelBinder.this.afterUnbindConsumer(destination, this.group, properties);
+					// TODO more error checking here
+					String recoverer = getErrorRecovererName(destination, group,
+							(ExtendedConsumerProperties<?>) properties);
+					if (getApplicationContext().containsBean(recoverer)) {
+						((DefaultSingletonBeanRegistry) getApplicationContext().getBeanFactory())
+							.destroySingleton(recoverer);
```

Comment:
I'm not sure if that will work - given that the binding is not a bean (which has been a discussion point for quite a while).